libraries/rush-lib/src/cli/parsing/SelectionParameterSet.ts

```diff
@@ -190,7 +191,10 @@ export class SelectionParameterSet {
    *
    * If no parameters are specified, returns all projects in the Rush config file.
    */
-  public async getSelectedProjectsAsync(terminal: ITerminal): Promise<Set<RushConfigurationProject>> {
+  public async getSelectedProjectsAsync(
+    terminal: ITerminal,
+    subspaceName?: string
```

Comment:
Instead of passing `subspaceName`, it would be more consistent for the constructor of `SelectionParameterSet` to accept an optional flag that tells it to create the special `--subspace` parameter.  The design seems to be that `SelectionParameterSet` creates and owns its parameters rather than receiving their values as inputs to `getSelectedProjectsAsync()`, and it seems to me that `--subspace` is really just another selection parameter (albeit degenerate because it is only supported by some commands and uses a different syntax).

Reply:
The main reason that `SelectionParameterSet` owns its own parameters is for DRY, since otherwise we were writing the same selection values over and over again.

Reply:
As of PR #4701, `getSelectedProjectsAsync()` calculates expressions using PNPM CLI operators like `--filter ...my-app`.  These selections will malfunction with subspaces. As a workaround, when subspaces are enabled the args get overwritten with a flat list.  For details, see https://github.com/microsoft/rushstack/pull/4707/commits/fb2c242388efa927cb560312f3056d4efd8d0889

In a future PR, Rush will need to (1) calculate the flat list of projects (without relying on PNPM operators) and (2) somehow communicate this list to `pnpm install` without exceeding the Windows maximum command length.